src/@providers/CONTRIBUTING.md

```diff
@@ -0,0 +1,197 @@
+    filterOptions?: Partial<ResourceOutputs['region']>,
+    pagingOptions?: Partial<PagingOptions>,
+  ): Promise<PagingResponse<ResourceOutputs['region']>> {
+    const {
+      data: { regions },
+    } = await this.client.region.listRegions({});
+    return {
+      total: regions.length,
+      rows: regions.map((region) => this.normalizeRegion(region)),
+    };
+  }
+}
+```
+
+### CRUD services
+
+```typescript
+export class MyVpcService extends CrudResourceService<'vpc', MyProviderCredentials> {
+  async get(id: string) {
+    // ...
+  }
+
+  async list(
+    filterOptions?: Partial<ResourceOutputs['vpc']>,
+    pagingOptions?: Partial<PagingOptions>,
+  ) {
+    // ...
+  }
+
+  create(
+    subscriber: Subscriber<string>,
+    inputs: ResourceInputs['vpc'],
+  ): Promise<ResourceOutputs['vpc']> {
+    // - Create new resource from inputs and return output values
+    // - Throw errors where necessary
+    // - Use subscriber.next('...') to share status updates with end-users
+  }
+
+  update(
+    subscriber: Subscriber<string>,
+    id: string,
+    inputs: DeepPartial<ResourceInputs['vpc']>,
+  ): Promise<ResourceOutputs['vpc']> {
+    // - Update existing resource matching the ID with new input values
+    // - Throw errors where necessary
+    // - Use subscriber.next('...') to share status updates with end-users
+  }
+
+  delete(subscriber: Subscriber<string>, id: string): Promise<void> {
+    // - Delete an existing resource matching the ID
+    // - Throw errors where necessary
+    // - Use subscriber.next('...') to share status updates with end-users
+  }
+}
+```
+
+### Terraform services
+
+Crud services are great for simple cloud operations that require only one or two API calls, but for more complex resources it can become tedious and error prone. To make this easier, arcctl services can leverage Terraform instead of defining individual `create`, `update`, and `delete` methods:
+
+```typescript
+export class MyService extends TerraformResourceService<'vpc', MyProviderCredentials> {
+  /**
+   * Version of terraform this service expects
+   */
+  readonly terraform_version = '1.4.5';
+
+  /**
+   * CDK Module to use for the service. See below for details.
+   */
+  readonly construct = MyServiceModule;
+
+  // ---
+  // Still needs the get and list functions for read flows
+  // ---
+
+  async get(id: string) {
+    // ...
+  }
+
+  async list(
+    filterOptions?: Partial<ResourceOutputs['vpc']>,
+    pagingOptions?: Partial<PagingOptions>,
+  ) {
+    // ...
+  }
+}
+```
+
+#### CDK Modules
+
+Modules contain all the resource creation logic for Terraform services. Modules are a thin wrapper around [CDK for Terraform](https://developer.hashicorp.com/terraform/cdktf). You can use all the features of `cdktf`, and the only requirement is that you conform to the input/output spec of the resource type in the associated service:
+
+```typescript
+export class MyVpcModule extends ResourceModule<'vpc', MyProviderCredentials> {
```

Comment:
If we want commands like `create` to work we should also showcase how to do `genImports` and the like.